### ISSUE TITLE
Question on statefile for the backend

### ISSUE THREAD

**Issue by rtomlinson-latacora** [cloudposse/terraform-aws-tfstate-backend#15]
Is there a requirement that the tfstate file for the backend resources needs to be part of the deployment that uses the backend? I don't see anything in the docs regarding this but whenever I deploy with the s3/dynamo backend, terraform always tried to destroy the s3 state bucket and dynamodb lock table. I wouldn't think they are necessarily coupled but maybe it's a requirement by terraform? Makes for very messy deletion because you're deleting backend resource at the same time as everything else.

**Comment by osterman** [comment]
@rtomlinson-latacora this sounds like maybe you're initializing the state backend, but then not importing the state. Please note this workflow is *different* from every other terraform module invocation. We have documented the process here: https://github.com/cloudposse/terraform-aws-tfstate-backend#usage and we've scripted it here: https://github.com/cloudposse/terraform-root-modules/blob/master/aws/tfstate-backend/scripts/init.sh
 
Basically, the process is you provision the backends using terraform but without remote state. Then you re-run `terraform init` with backends configured and it will prompt to import the backend state to S3.

**Comment by rtomlinson-latacora** [comment]
@osterman If I'm understanding this correctly, the point of the script is to make the backend resources part of state. That is effectively what the second `terraform init` is doing. All resources that you later add, will also be a part of this state. You have one state file which has the backend resources plus anything else you add to your deployments.

**Comment by osterman** [comment]
That's sounds correct. One backend can be shared by multiple projects. Just use different key prefixes (e.g. `us-west-2/vpc/terraform.tfstate`). We recommend using one bucket per AWS account. Also, for true terraform-HA don't even share the bucket across regions.

**Comment by rtomlinson-latacora** [comment]
Okay, I think I'm sorta understanding how you would namespace the other backends. I'll just do a little experiment to confirm. Since you're storing the state file that manages the dynamodb and bucket for the backend itself in said bucket, you'd obviously run into problems when trying to delete the backend. And since deleting the backend shouldn't need to occur often I'm assuming that this is just a management cost that you'd have to deal with in the event that you actually need to do something with the backend.

**Comment by osterman** [comment]
@rtomlinson-latacora here's the process of destroying the bucket: https://github.com/cloudposse/terraform-root-modules/blob/master/aws/tfstate-backend/scripts/destroy.sh

**Comment by osterman** [comment]
@rtomlinson-latacora thanks for reporting your issues with the coldstart process and destruction. If you have further issues, please let us know! I'm going to close this issue for now.
